tools/buildsys/src/args.rs

```diff
@@ -165,17 +198,23 @@ fn sensitive_env_vars(build_type: BuildType) -> impl Iterator<Item = &'static st
 }
 
 /// Emits the cargo directives for a the list of sensitive environment variables for a given
-/// `[BuildType]`.
+/// `[BuildType]`, unless the `[BuildType]` is Repack.
 pub(crate) fn rerun_for_envs(build_type: BuildType) {
-    for var in sensitive_env_vars(build_type) {
-        println!("cargo:rerun-if-env-changed={}", var)
+    match build_type {
+        BuildType::Repack => (),
+        _ => {
+            for var in sensitive_env_vars(build_type) {
+                println!("cargo:rerun-if-env-changed={}", var)
+            }
+        }
     }
 }
 
 /// The thing that buildsys is building.
 #[repr(u8)]
 #[derive(Debug, Clone, Copy)]
 pub(crate) enum BuildType {
+    Repack = 0b00000000,
     Package = 0b00000001,
     Variant = 0b00000010,
```

Comment:
Seems like this is an issue. Zero is supposed to mean "no flags are set" but you have assigned one of the flags to zero. Maybe this would fix the cargo rerun output problem.

```suggestion
    Repack = 0b00000001,
    Package = 0b00000010,
    Variant = 0b00000100,
```